### DISCUSSION TITLE
MyBMW 1.5.0 app connects to the car without apps until you run it.

### DISCUSSION THREAD

**Discussion by unlive74** [BimmerGestalt/AAIdrive#346] [General]
Today the MyBMW app has been updated to version 1.5.0 and there are new problems - it connects to the car without apps until you run it. I downgraded to 1.4.3 - it still works well.

**Comment by Marc2480** [comment]
I got the update to 1.5.0 yesterday, running AAI 1.3rc2-43. All working as previously, even restarted my S20 to make sure, and even then it all connected ;)

**Comment by hufman** [comment]
I also experienced this lack of apps, which was surprising because it has been very reliable for me lately, and I discovered that (the Mini version of) MyBMW had a new screen of EULAs for me to click through. I do see now that I've indeed been upgraded to 1.5.0, but the connection has been solid for me since that first time.
Did you also see this EULA screen once, and does it continue to work afterward like it does for me?

**Reply by Marc2480** [reply]
Hi Walter, indeed, I recall seeing this one after the upgrade to 1.5.0 briefly, must have confirmed and forgotten ...

**Comment by unlive74** [comment]
I confirmed the agreement right after installation. Connection problem was 2 times out of 4 starts. After that, I downgraded the application.

@hufman if necessary, I can try to send a connection log without apps.

**Reply by hufman** [reply]
Did the car show the Apps checkbox as filled in before you open the MyBMW app? If not, then there is a bug in the automatic connection logic in MyBMW and you should use the feedback form to report it to BMW. As a very small minority of users, they need to know of any bugs that impact this feature!

**Comment by fabssor** [comment]
Just feedback from me. I do not have any problems with the new My BMW Version. However I am living in Germany and therefore running the European version.

**Reply by drm87** [reply]
I can confirm that. In my case it seems now to even work over USB, without needing the old Connected-App to be installed.

But it still puzzles me, as without AAID there would now be no apps that would use the connection. Great for us, but strange decision from BMW (or are they planning to introduce new apps in MyBMW, which i doubt?).

**Reply by hufman** [reply]
[MyBMW 1.5.0](https://play.google.com/store/apps/details?id=de.bmw.connected.mobile20.na) claims to add `Display of vehicle status (mileage, fuel level, range) now available for some older vehicle generations via Bluetooth/USB` in the What's New section.
After version 1.2 re-enabled the BMW Apps support, I had suspected this was the eventual goal. MyBMW 1.2 and later includes one built-in car app named "cdsbaseapp", and CDS would mean the [Car Data Service](https://hufman.github.io/BMWConnectedAnalysis/cds/) that can provide this vehicle status directly through the car app protocol. The CDS is also key to the Alexa integration in new cars. I believe a lot of the initial motivation for stabilizing the BT connection was because MyBMW uses a related protocol over BT to obtain the Android Auto wifi password from ID7. By officially claiming support for CDS information in the changelog, this probably means they're confident that they've stabilized the BT and USB app connection.

I don't know what to expect, about whether they plan to add other apps for this (finally reliable) connection. I'd love to see them bring back the Calendar, which was one of the neatest integrations, and the Mini FindMate system is an expensive add-on that is useless without the Mini Connected app. Their new Share From Google Maps functionality would need an RHMI app to trigger navigation in cars that don't have cloud connectivity. I don't expect them to bring back the Car API SDK 3rd-party apps, because the SDK code was removed from MyBMW: I believe they are doing some work with SmartDeviceLink/AppLink for similar app remoting, but I don't think this will filter down below ID7.
MyBMW does contain a database to dynamically add/remove/enable/disable car apps, so it would be entirely possible (and easy) for them to build in any extra car app modules and even offer configuration options in the new Flutter UI. I'm personally impressed with the rate of improvement I've seen in this new app, and I hope it continues.

**Reply by drm87** [reply]
Thanks for the detailed update on your findings! 😃 That looks promising, I think.
